src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -1024,16 +1026,20 @@ void EmitAlternation(RegexNode node)
                 // the whole alternation can be treated as a simple switch, so we special-case that. However,
                 // we can't goto _into_ switch cases, which means we can't use this approach if there's any
                 // possibility of backtracking into the alternation.
-                bool useSwitchedBranches = isAtomic;
-                if (!useSwitchedBranches)
+                bool useSwitchedBranches = false;
+                if ((node.Options & RegexOptions.RightToLeft) == 0)
```

Comment:
This should strictly only be true for a lookbehind in a left-to-right pattern, right? I ask because if there are any other cases where we are now planning to actually support right-to-left, we would need to modify the Scan() override to emit code that moves in the right direction.

Reply:
If (node.Options & RegexOptions.RightToLeft) == 0, it's not a lookbehind.  Basically there's just one concept: process forward or backward, dictated by whether the node has RightToLeft set.  If you're in a lookbehind, the nodes it contains will all be RightToLeft.

Reply:
But, yes, once we enable the top-level RightToLeft option, at that point we also need to be able to make the bumpalong process backwards.